[
   {
      "author_association" : "NONE",
      "body" : "Hey, I would like to work on this issue, do you want this https://github.com/bitcoin/bitcoin/blob/master/contrib/signet/getcoins.py#L123 (except block) to handle exception only if the args.captcha couldn't fetched by session.get [requests.exceptions.ConnectionError]? and also I think we can use https://wiki.python.org/moin/EscapingHtml to santize inputs.",
      "created_at" : "2022-01-21T14:33:20Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/24119#issuecomment-1018555832",
      "id" : 1018555832,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/24119",
      "node_id" : "IC_kwDOABII5848te24",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1018555832/reactions"
      },
      "updated_at" : "2022-01-21T14:39:39Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1018555832",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/58676502?v=4",
         "events_url" : "https://api.github.com/users/sh15h4nk/events{/privacy}",
         "followers_url" : "https://api.github.com/users/sh15h4nk/followers",
         "following_url" : "https://api.github.com/users/sh15h4nk/following{/other_user}",
         "gists_url" : "https://api.github.com/users/sh15h4nk/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/sh15h4nk",
         "id" : 58676502,
         "login" : "sh15h4nk",
         "node_id" : "MDQ6VXNlcjU4Njc2NTAy",
         "organizations_url" : "https://api.github.com/users/sh15h4nk/orgs",
         "received_events_url" : "https://api.github.com/users/sh15h4nk/received_events",
         "repos_url" : "https://api.github.com/users/sh15h4nk/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/sh15h4nk/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/sh15h4nk/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/sh15h4nk"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "> do you want this https://github.com/bitcoin/bitcoin/blob/master/contrib/signet/getcoins.py#L123 (except block) to handle exception only if the args.captcha couldn't fetched by session.get [requests.exceptions.ConnectionError]?\r\n\r\n1. Yes we need to improve that `except` block\r\n2. Based on my understanding we need multiple `except` blocks to handle different types of exceptions. I am not sure hence created an issue.\r\n\r\n>and also I think we can use https://wiki.python.org/moin/EscapingHtml to sanitize inputs.\r\n\r\n1. Sanitize image\r\n2. Require captcha to be in a specific format\r\n3. Limit size of image\r\n\r\nCan also add one warning or note below captcha that this is based on faucet website URL.",
      "created_at" : "2022-01-21T15:26:05Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/24119#issuecomment-1018609387",
      "id" : 1018609387,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/24119",
      "node_id" : "IC_kwDOABII5848tr7r",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1018609387/reactions"
      },
      "updated_at" : "2022-01-21T15:26:05Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1018609387",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/13405205?v=4",
         "events_url" : "https://api.github.com/users/prayank23/events{/privacy}",
         "followers_url" : "https://api.github.com/users/prayank23/followers",
         "following_url" : "https://api.github.com/users/prayank23/following{/other_user}",
         "gists_url" : "https://api.github.com/users/prayank23/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/prayank23",
         "id" : 13405205,
         "login" : "prayank23",
         "node_id" : "MDQ6VXNlcjEzNDA1MjA1",
         "organizations_url" : "https://api.github.com/users/prayank23/orgs",
         "received_events_url" : "https://api.github.com/users/prayank23/received_events",
         "repos_url" : "https://api.github.com/users/prayank23/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/prayank23/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/prayank23/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/prayank23"
      }
   },
   {
      "author_association" : "NONE",
      "body" : "Hi, I would like to contribute here. I think I can do something about the latter issue. I just want more clarity regarding the former one: I have one question: do all the `except` block print the same error output?\r\nRe:\r\n> Except block directly handles BaseException. Handling 'BaseException' means that system exits and keyboard interrupts may be mis-handled.\r\n\r\nDoes this mean that we want to take care of the case when there is an error generated by keyboard interrupts? And is this the only error we want to take special care of?",
      "created_at" : "2022-01-25T14:42:01Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/24119#issuecomment-1021253401",
      "id" : 1021253401,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/24119",
      "node_id" : "IC_kwDOABII58483xcZ",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1021253401/reactions"
      },
      "updated_at" : "2022-01-25T14:42:01Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1021253401",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/67277428?v=4",
         "events_url" : "https://api.github.com/users/akshatdalton/events{/privacy}",
         "followers_url" : "https://api.github.com/users/akshatdalton/followers",
         "following_url" : "https://api.github.com/users/akshatdalton/following{/other_user}",
         "gists_url" : "https://api.github.com/users/akshatdalton/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/akshatdalton",
         "id" : 67277428,
         "login" : "akshatdalton",
         "node_id" : "MDQ6VXNlcjY3Mjc3NDI4",
         "organizations_url" : "https://api.github.com/users/akshatdalton/orgs",
         "received_events_url" : "https://api.github.com/users/akshatdalton/received_events",
         "repos_url" : "https://api.github.com/users/akshatdalton/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/akshatdalton/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/akshatdalton/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/akshatdalton"
      }
   }
]
